student.rb

```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
```

Comment:
Thank you for the review!
According to rubocop it seems I should [use keywords for boolean values](https://docs.rubocop.org/rubocop/1.13/cops_style.html#styleoptionalbooleanparameter).
I got some info about this from [this answer](https://stackoverflow.com/questions/67391214/style-optionalbooleanparameter-use-keyword-arguments-when-defining-method-with).

So in this case I will need to use it like 
`super(age, name, parent_permission:)`

I hope this is an acceptable solution, please tell me if it doesn't fit.

Reply:
Hello @mohamedSabry0, Yes it's an acceptable solution

Reply:
Unfortunately, there are some errors:
- this style: `super(age, name, parent_permission:)` produces the following error:
 ```ruby
    student.rb:3:40: F: Lint/Syntax: unexpected token tRPAREN
    (Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
    super(age, name, parent_permission:)
```
**For some reason, I don't see an error on my local machine running Rubocop. I did try to update the `.rubocop.yml` but the error still appears on Github actions**

- this style `super(age, name, parent_permission: parent_permission)` produces the following error:
```ruby
    student.rb:3:41: C: [Correctable] Style/HashSyntax: Omit the hash value.
    super(age, name, parent_permission: parent_permission)
```
**This one appears locally only. I'm not sure why**

- this style: `super(age, name, parent_permission)` along with `initialize(age, classroom, name = 'Unknown', parent_permission = true)` produces the following error:
 ```ruby
   student.rb:2:52: C: Style/OptionalBooleanParameter: Prefer keyword arguments for 
   arguments with a boolean default value; 
   use parent_permission: true instead of parent_permission = true.
   def initialize(age, classroom, name = 'Unknown', parent_permission = true)
```
After all the trials, I think the second one doesn't violate the live version of Rubocop on Github.
So should I go with this or disable a cop?